src/compose.ts

```diff
@@ -44,18 +44,57 @@ export default function compose<R>(
 export default function compose<R>(...funcs: Function[]): (...args: any[]) => R
 
 export default function compose(...funcs: Function[]) {
+  // infer the argument type so it is usable in inference down the line
   if (funcs.length === 0) {
-    // infer the argument type so it is usable in inference down the line
     return <T>(arg: T) => arg
   }
-
   if (funcs.length === 1) {
     return funcs[0]
   }
-
   return funcs.reduce(
     (a, b) =>
       (...args: any) =>
         a(b(...args))
   )
 }
+
+/*
+  #SCC
+  How are there more than one functions with the same name?
+  "That's TS method overloading. One actual code function, many type signatures with different forms of the arguments."
+   - Mark Erikson https://twitter.com/acemarke/status/1549622836926431232
+
+  According to Wikipedia,
+  Function overloading or method overloading is the ability to create multiple functions of the same name with different implementations
+
+  Do's and Don'ts for overloads:
+  https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html#overloads-and-callbacks
+
+  When this code is compiled to JavaScript, the concrete function alone will be visible.
+  When TS compiles this file it turns it into this JavaSript:
+  function compose(...funcs) {
+    if (funcs.length === 0) { return (arg) => arg; }
+    if (funcs.length === 1) { return funcs[0]; }
+    return funcs.reduce((a, b) => (...args) => a(b(...args)));
+  }
+
+  So compose() seems to be a utility function that allows you to do this compose(f, g, h)
+  when you would otherwise have needed to this (...args) => f(g(h(...args)))
+
+  From https://github.com/reduxjs/redux/blob/master/docs/api/compose.md:
+  Composes functions from right to left.
+
+  This is a functional programming utility, and is included in Redux as a convenience.
+  You might want to use it to apply several store enhancers in a row.
+
+  Arguments (arguments): The functions to compose. Each function is expected to accept a single parameter.
+  Its return value will be provided as an argument to the function standing to the left, and so on.
+  The exception is the right-most argument which can accept multiple parameters,
+  as it will provide the signature for the resulting composed function.
+  Returns (Function): The final function obtained by composing the given functions from right to left.
+
+  All compose does is let you write deeply nested function transformations without the rightward drift of the code.
+
+  I understand what the compose function does but I am still not sure how to use the overloading.
+  - Seth Broweleit
+*/
```

Comment:
Just adding a bit of context here for folks who maybe never used Redux before the `compose` utility was added, aka pre-Redux Toolkit. 

Redux has always been criticized for the amount of boilerplate code that must be written when setting up a Redux state store. This was valid criticism that Redux Toolkit (aka Redux 2.0) was designed to solve. One hangup from vanilla Redux was the application of middleware / [Store Enhancers](https://teams.microsoft.com/l/entity/com.microsoft.teamspace.tab.wiki/tab::3205c02d-c8bb-46fc-b47e-c3c64fb05100?context=%7B%22subEntityId%22%3A%22%7B%5C%22pageId%5C%22%3A8%2C%5C%22sectionId%5C%22%3A14%2C%5C%22origin%5C%22%3A2%7D%22%2C%22channelId%22%3A%2219%3Apba6flnxCO70wNgOsxLvA9n6BAKut4aC7CQDkoWVbUU1%40thread.tacv2%22%7D&tenantId=6c637512-c417-4e78-9d62-b61258e4b619) to your code. The most important middlewares, `thunk`, for async reducers, and the browser dev tools, took quite a bit of code to setup and was generally a drag. `compose` allows us to simply pass those store enhancers to the `createStore` function as a parameter and those functions are [curried](https://stackoverflow.com/questions/36314/what-is-currying) and their result applied to the `store` object in a predictable way. 

The docs say that we will likely never write a store enhancer and that's probably true. That said, the simplified way of setting up our store with `compose` is very helpful. I don't think memorizing the internals of `compose` is necessary to work with Redux because you'll generally interact with the function is the same way each time you create a new store. (And you only do that once in any application.) 

That said, I think the overloading pattern is really cool and gives me yet another reason to try to learn more functional programming!

Reply:
I should have read this comment before adding my comment above LOL Oh well, cool to start to piece some of this together. Although I have a long way to go before I have a thorough comprehension of this library LOL